anycable-go/templates/_helpers.tpl

```diff
@@ -41,3 +41,291 @@ Template to generate secrets for a private Docker repository for K8s to use
+    {{ .Values.global.anycableGo.env.anycableMetricsLogInterval | quote }}
+{{- else if not (empty .Values.env.anycableMetricsLogInterval) -}}
+    {{ .Values.env.anycableMetricsLogInterval | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Template to provide a value for ANYCABLE_METRICS_LOG_FORMATTER
+*/}}
+{{- define "anycableGo.anycableMetricsLogFormatter" -}}
+{{- if not (empty .Values.global.anycableGo.env.anycableMetricsLogFormatter) -}}
+    {{ .Values.global.anycableGo.env.anycableMetricsLogFormatter | quote }}
+{{- else if not (empty .Values.env.anycableMetricsLogFormatter) -}}
+    {{ .Values.env.anycableMetricsLogFormatter | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Template to provide a value for ANYCABLE_METRICS_HTTP
+*/}}
+{{- define "anycableGo.anycableMetricsHttp" -}}
+{{- if not (empty .Values.global.anycableGo.env.anycableMetricsHttp) -}}
+    {{ .Values.global.anycableGo.env.anycableMetricsHttp | quote }}
+{{- else if not (empty .Values.env.anycableMetricsHttp) -}}
+    {{ .Values.env.anycableMetricsHttp | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Template to provide a value for ANYCABLE_METRICS_HOST
+*/}}
+{{- define "anycableGo.anycableMetricsHost" -}}
+{{- if not (empty .Values.global.anycableGo.env.anycableMetricsHost) -}}
+    {{ .Values.global.anycableGo.env.anycableMetricsHost | quote }}
+{{- else if not (empty .Values.env.anycableMetricsHost) -}}
+    {{ .Values.env.anycableMetricsHost | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Template to provide a value for ANYCABLE_METRICS_PORT
+*/}}
+{{- define "anycableGo.anycableMetricsPort" -}}
+{{- if not (empty .Values.global.anycableGo.env.anycableMetricsPort) -}}
+    {{ .Values.global.anycableGo.env.anycableMetricsPort | quote }}
+{{- else if not (empty .Values.env.anycableMetricsPort) -}}
+    {{ .Values.env.anycableMetricsPort | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Template to provide a value for ANYCABLE_HEALTH_PATH
+*/}}
+{{- define "anycableGo.anycableHealthPath" -}}
+{{- if not (empty .Values.global.anycableGo.env.anycableHealthPath) -}}
+    {{ .Values.global.anycableGo.env.anycableHealthPath | quote }}
+{{- else if not (empty .Values.env.anycableHealthPath) -}}
+    {{ .Values.env.anycableHealthPath | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Template to provide a value for ANYCABLE_MAX_MESSAGE_SIZE
+*/}}
+{{- define "anycableGo.anycableMaxMessageSize" -}}
+{{- if not (empty .Values.global.anycableGo.env.anycableMaxMessageSize) -}}
+    {{ .Values.global.anycableGo.env.anycableMaxMessageSize | quote }}
+{{- else if not (empty .Values.env.anycableMaxMessageSize) -}}
+    {{ .Values.env.anycableMaxMessageSize | quote }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Template to provide a value for ANYCABLE_REDIS_URL
+*/}}
+{{- define "anycableGo.anycableRedisUrl" -}}
+{{- if not (empty .Values.global.anycableGo.redisUrlSecret) -}}
+secretKeyRef:
+    name: {{ .Values.global.anycableGo.redisUrlSecret.name }}
+    key: {{ .Values.global.anycableGo.redisUrlSecret.key }}
```

Comment:
I'm not sure about this.
Hey, @mmuehlberger, could you clarify why do you need customization of a key here?

I understand how it works, but it looks to me that it adds complexity without bare necessity.
I think that integration of a secret to pod/deployment is not a part of the chart's interface. The only thing we should care about is how to get values into the secret, not how to keep it inside.

Reply:
My guess @mmuehlberger may have a situation where he has an externally controlled secret which has a number of secrets including redis ones. But it is only a possible situation.